### ISSUE TITLE
Django ckeditor fields not loaded via Add Another button

### ISSUE THREAD

**Issue by geolimber** [theatlantic/django-nested-admin#175]
Hi. I have a problem using nested admin with inline models that have ckeditor fields.
I have models like this: Game, Question, Hint. 
Game could have a lot of questions, every question have 3 hints. 
Using builtin inlines in django admin there is no such problem. I can click Add Another Question and form loads with ckeditor text field. But i cant put here a hint inline.
Using django-nested-admin i can put inline hints inside question, but clicking Add Another Question doesn't load ckeditor form fields, this fields loads as text fields.
Wich way i should dig?
Thanks

**Comment by andrkhr** [comment]
I have the same problem. But the fields load normally if you use the "+" button on inline form

**Comment by fdintino** [comment]
I see two things in ckeditor-init.js that would cause these issues. The first is that django-ckeditor registers a click handler, instead of using the jQuery `formset:added` event that the admin fires. In this case, the selectors for the click event handler don't match up exactly with django-nested-admin:

https://github.com/django-ckeditor/django-ckeditor/blob/155e80ad84de88b4bf0570d1afe7cae2c2f6e20d/ckeditor/static/ckeditor/ckeditor-init.js#L56-L65

The other possible issue has to do with the onload initializer in that file.

https://github.com/django-ckeditor/django-ckeditor/blob/155e80ad84de88b4bf0570d1afe7cae2c2f6e20d/ckeditor/static/ckeditor/ckeditor-init.js#L41-L54

Specifically the line:

```javascript
if (t.getAttribute('data-processed') == '0' && t.id.indexOf('__prefix__') == -1) {
```

Because 'empty' forms (i.e. forms whose ids are `id_prefix-empty`) can have nested children, it's possible for event handlers to get double registered. I haven't verified if that's happening here, but adding this additional conditional should help:

```javascript
if (... && t.id.indexOf('-empty-') === -1) {
```

A version of `ckeditor-init.js` from django-ckeditor that ought to work with django-nested-admin would be:

```javascript
/* global CKEDITOR */
;(function() {
  var el = document.getElementById('ckeditor-init-script');
  if (el && !window.CKEDITOR_BASEPATH) {
    window.CKEDITOR_BASEPATH = el.getAttribute('data-ckeditor-basepath');
  }

  // Polyfill from https://developer.mozilla.org/en/docs/Web/API/Element/matches
  if (!Element.prototype.matches) {
    Element.prototype.matches =
        Element.prototype.matchesSelector ||
        Element.prototype.mozMatchesSelector ||
        Element.prototype.msMatchesSelector ||
        Element.prototype.oMatchesSelector ||
        Element.prototype.webkitMatchesSelector ||
        function(s) {
            var matches = (this.document || this.ownerDocument).querySelectorAll(s),
                i = matches.length;
            while (--i >= 0 && matches.item(i) !== this) {}
            return i > -1;
        };
  }

  function runInitialisers() {
    if (!window.CKEDITOR) {
      setTimeout(runInitialisers, 100);
      return;
    }

    initialiseCKEditor();
    initialiseCKEditorInInlinedForms();
  }

  if (document.readyState != 'loading' && document.body) {
    document.addEventListener('DOMContentLoaded', initialiseCKEditor);
    runInitialisers();
  } else {
    document.addEventListener('DOMContentLoaded', runInitialisers);
  }

  function initialiseCKEditor() {
    var textareas = Array.prototype.slice.call(document.querySelectorAll('textarea[data-type=ckeditortype]'));
    for (var i=0; i<textareas.length; ++i) {
      var t = textareas[i];
      if (t.getAttribute('data-processed') == '0' && t.id.indexOf('__prefix__') == -1 && t.id.indexOf('-empty-') === -1) {
        t.setAttribute('data-processed', '1');
        var ext = JSON.parse(t.getAttribute('data-external-plugin-resources'));
        for (var j=0; j<ext.length; ++j) {
          CKEDITOR.plugins.addExternal(ext[j][0], ext[j][1], ext[j][2]);
        }
        CKEDITOR.replace(t.id, JSON.parse(t.getAttribute('data-config')));
      }
    }
  }

  function initialiseCKEditorInInlinedForms() {
    document.body.addEventListener('click', function(e) {
      if (e.target && (
        e.target.matches('.add-row a') ||
        e.target.matches('.grp-add-handler')
      )) {
        initialiseCKEditor();
      }
    });

    if (typeof django === 'object' && django.jQuery) {
      django.jQuery(document).on('formset:added', initialiseCKEditor);
    }
  }

}());
```

You can effectively override that static file, without needing to fork or modify django-ckeditor directly, if you make a custom `INSTALLED_APP` (let's call it `ckeditor_customizations`), have it precede `'ckeditor'` in the `INSTALLED_APPS` list, and place the contents above in:

`your_project/ckeditor_customizations/static/ckeditor/ckeditor-init.js`

**Comment by geolimber** [comment]
> I see two things in ckeditor-init.js that would cause these issues. The first is that django-ckeditor registers a click handler, instead of using the jQuery `formset:added` event that the admin fires. In this case, the selectors for the click event handler don't match up exactly with django-nested-admin:
> 
> https://github.com/django-ckeditor/django-ckeditor/blob/155e80ad84de88b4bf0570d1afe7cae2c2f6e20d/ckeditor/static/ckeditor/ckeditor-init.js#L56-L65
> 
> The other possible issue has to do with the onload initializer in that file.
> 
> https://github.com/django-ckeditor/django-ckeditor/blob/155e80ad84de88b4bf0570d1afe7cae2c2f6e20d/ckeditor/static/ckeditor/ckeditor-init.js#L41-L54
> 
> Specifically the line:
> 
> ```js
> if (t.getAttribute('data-processed') == '0' && t.id.indexOf('__prefix__') == -1) {
> ```
> ....

Unfortunantely this solution does not make any effect. Still struggling whit correct loading ckeditor widget hitting "Add another" button. I tryed override and even replace this ckeditor-init.js by one provided in your solution. 
![screen_04_08_2020-21:10:40](https://user-images.githubusercontent.com/35577127/89360733-ddb25580-d6b8-11ea-9e60-5463979ab7ca.png)

**Comment by fdintino** [comment]
I suspect you neglected to also add the `formset:added` event handler from my comment. I'll open a pull request against django-ckeditor that makes this change.

**Comment by fdintino** [comment]
This is fixed in the 6.0.0 release of django-ckeditor.
